dali/base/dacoven.cpp

```diff
@@ -344,6 +329,23 @@ CIArrayOf<CDaliUidAllocator> CCovenBase::foreginUidallocators;
 CDaliUidAllocator CCovenBase::localUidAlloctor;
 
 
+CDaliUidAllocator &CDaliUidAllocator::find(CIArrayOf<CDaliUidAllocator> &uidallocators,const SocketEndpoint &foreignnode)
+{
+    CriticalBlock block(CCovenBase::uidcrit);
+    ForEachItemIn(i,uidallocators) {
+        if (uidallocators.item(i).node.equals(foreignnode))
+            return uidallocators.item(i);
+    }
+    CDaliUidAllocator &ret = *new CDaliUidAllocator(foreignnode);
+    uidallocators.append(ret);
+    if (uidallocators.ordinality()>1) {
```

Comment:
Probably not a biggie but this could be out of the critblock

Reply:
that highlights something else, the logic here was weird, it assumed that 1st added was 'local' and any others were foreign.
But with this change any call to find is looking for a foreign uidallocator, so this condition (>1), shouldn't be here either.
I'll change.